samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -118,11 +118,30 @@ def on_before_transform_template(self, template_dict):
                     # Lazy initialization of the client- create it when it is needed
                     if not self._sar_client:
                         self._sar_client = boto3.client("serverlessrepo")
-                    service_call(app_id, semver, key, logical_id)
+                    self._make_service_call_with_retry(service_call, app_id, semver, key, logical_id)
                 except InvalidResourceException as e:
                     # Catch all InvalidResourceExceptions, raise those in the before_resource_transform target.
                     self._applications[key] = e
 
+    def _make_service_call_with_retry(self, service_call, app_id, semver, key, logical_id):
+        call_succeeded = False
+        start_time = time()
+        while (time() - start_time) < self.TEMPLATE_WAIT_TIMEOUT_SECONDS:
```

Comment:
There's also a similar retry pattern used in `on_after_transform_template`. I can see things are done a bit differently within this retry and that retry. But can we try to refactor the retry part into a decorator that can be used on both?